docker/README.md

```diff
@@ -0,0 +1,49 @@
+# FIWARE IoT Agent for the Ultralight 2.0 protocol
+
+[![FIWARE IoT Agents](https://nexus.lab.fiware.org/repository/raw/public/badges/chapters/iot-agents.svg)](https://www.fiware.org/developers/catalogue/)
+[![](https://nexus.lab.fiware.org/repository/raw/public/badges/stackoverflow/iot-agents.svg)](https://stackoverflow.com/questions/tagged/fiware+iot)
+
+An Internet of Things Agent for the Ultralight 2.0 protocol (with
+[AMQP](https://www.amqp.org/), [HTTP](https://www.w3.org/Protocols/) and
+[MQTT](https://mqtt.org/) transports). This IoT Agent is designed to be a bridge
+between Ultralight and the
+[NGSI](https://swagger.lab.fiware.org/?url=https://raw.githubusercontent.com/Fiware/specifications/master/OpenAPI/ngsiv2/ngsiv2-openapi.json)
+interface of a context broker.
+
+It is based on the
+[IoT Agent Node.js Library](https://github.com/telefonicaid/iotagent-node-lib).
+Further general information about the FIWARE IoT Agents framework, its
+architecture and the common interaction model can be found in the library's
+GitHub repository.
+
+This project is part of [FIWARE](https://www.fiware.org/). For more information
+check the FIWARE Catalogue entry for the
+[IoT Agents](https://github.com/Fiware/catalogue/tree/master/iot-agents).
+
+## Configuration with environment variables
+Some of the more common variables can be configured using environment variables. The ones overriding general parameters
+in the `config.iota` set are described in the [IoTA Library Configuration manual](https://github.com/telefonicaid/iotagent-node-lib#configuration).
+
+The ones specifically relating to Ultralight bindings are described in the following table.
```

Comment:
Link added. Even if the table goes stale, the explanatory text directs you onwards for the latest information. Fixed fa9481c

Reply:
Putting myself in developers's shoes, I prefer to navigate a link to get the accurate list of settings (please, don't assume our target developers are so lazy... they use to be smart people ;) that spending hours to find a problem due to the env var which name I though was correct actually it wasn't.

In fact, a link is already added for library env vars. Why don't do the same for agent-specific env vars?

With regards to:

> Most good Docker Hub README files list the ENV variables you can use, we should attempt to follow good practice.

Do these projects duplicate env vars information (in Docker hub and upstream software documentation)? Or they only have the copy of the env vars in the Docker hub?

I think it would be a good idea to gets more opinions on this. @dcalvoalonso @AlvaroVega what do you think?

Reply:
+1 to @fgalan, a link should be enough

Reply:
+1 to @fgalan, repeated information is more difficult to maintain and at the end, something is not updated and causes more problems than a single link

Reply:
Documentation usability is negatively affected if you need to click through to a greater depth, assuming someone looks up Docker information on Docker Hub (which is a fair assumption) and **fails to find it there**, effectively you are saying:

## [Thank you Mario, but our princess is in another castle](https://knowyourmeme.com/memes/but-our-princess-is-in-another-castle)

Which is mildly annoying. 😠  - Why do we want to annoy our developers?

> Do these projects duplicate env vars information (in Docker hub and upstream software documentation)? Or they only have the copy of the env vars in the Docker hub?

As an example of Good practice, look at the following for **MySQL**:

*  The real documentation - https://dev.mysql.com/doc/refman/8.0/en/docker-mysql-more-topics.html#docker-environment-variables
*  The equivalent on Docker Hub:  https://hub.docker.com/_/mysql/

This to me looks like a reasonable compromise on this would be to include an opening paragraph with basic docker usage,  a prominent link in it and at least the major configuration elements in it. I'm making the assumption here that the major `ENV` variables are unlikely to be renamed.

Reply:
If we restrict the list of env vars to be described to the ones in the docker-compose.yml used as example fine with me. In fact, this will be a way of documenting the docker-compose.yml embedded in the README.md. That is:

```
        - "IOTA_CB_HOST=orion"
        - "IOTA_CB_PORT=1026"
        - "IOTA_NORTH_PORT=4041"
        - "IOTA_REGISTRY_TYPE=mongodb"
        - "IOTA_MONGO_HOST=mongo-db"
        - "IOTA_MONGO_PORT=27017"
        - "IOTA_MONGO_DB=iotagent-ul"
        - "IOTA_HTTP_PORT=7896"
        - "IOTA_PROVIDER_URL=http://iot-agent:4041"
```

In addition, I'd suggest not using the current table style (which maps env var with config.js element) but a short description for each one, in the same fashion that the https://hub.docker.com/_/mysql/ you cite.

If others develpers are ok with this (@dcalvoalonso @AlvaroVega what do you think?) please go ahead.

Reply:
If only some of the env vars are going to be in that page I'd prefer not use a table, just a short description.

Reply:
@fgalan @AlvaroVega  - a limited bullet list sounds like a good idea - a list appears to be less **comprehensive** than a formal table, but still provides sufficient documentation to be able to run a simple dockerized solution.

Implicitly it provides the following information:

1.  The system can be configured using Environment variables
2.  Some major environment variables are documented here on **DockerHub** - we are able to write a good README 
3.  Environment variables are documented in more detail **elsewhere** (avoid duplicate effort)
4.  You won't find **everything** here, but we are considerate enough to get you started and provide a link.

In summary this approach states -  _we aim to be nice to the developer._

Reply:
@dcalvoalonso what dou you think?

Reply:
Duplicate Table replaced with shortened list and sign-posted links to the actual documentation.

Fixed - 92fe3e19